src/background/deploymentUpdater.ts

```diff
@@ -88,146 +88,153 @@ const locateAllForService = locator.locateAllForService.bind(locator);
+  const activatedModComponents = selectActivatedModComponents({
+    options: optionsState,
+  });
 
-  const toUninstall = installed.filter(
-    (extension) => !isEmpty(extension._deployment),
+  const modComponentsToDeactivate = activatedModComponents.filter(
+    (activatedModComponent) => !isEmpty(activatedModComponent._deployment),
   );
 
-  if (toUninstall.length === 0) {
+  if (modComponentsToDeactivate.length === 0) {
     // Short-circuit to skip reporting telemetry
     return;
   }
 
-  await uninstallExtensionsAndSaveState(toUninstall, {
+  await deactivateModComponentsAndSaveState(modComponentsToDeactivate, {
     editorState,
     optionsState,
   });
 
   reportEvent("DeploymentDeactivateAll", {
     auto: true,
-    deployments: toUninstall.map((x) => x._deployment.id),
+    deployments: modComponentsToDeactivate.map((x) => x._deployment.id),
   });
 }
 
-async function uninstallUnmatchedDeployments(
-  deployments: Deployment[],
+async function deactivateUnassignedDeployments(
+  assignedDeployments: Deployment[],
 ): Promise<void> {
   const [optionsState, editorState] = await Promise.all([
     getModComponentState(),
     getEditorState(),
   ]);
-  const installed = selectActivatedModComponents({ options: optionsState });
+  const activatedModComponents = selectActivatedModComponents({
+    options: optionsState,
+  });
 
-  const deploymentRecipeIds = new Set(
-    deployments.map((deployment) => deployment.package.package_id),
+  const deployedModIds = new Set(
+    assignedDeployments.map((deployment) => deployment.package.package_id),
   );
 
-  const toUninstall = installed.filter(
-    (extension) =>
-      !isEmpty(extension._deployment) &&
-      !deploymentRecipeIds.has(extension._recipe?.id),
+  const modComponentsToDeactivate = activatedModComponents.filter(
+    (activatedModComponent) =>
+      !isEmpty(activatedModComponent._deployment) &&
+      !deployedModIds.has(activatedModComponent._recipe?.id),
   );
 
-  if (toUninstall.length === 0) {
+  if (modComponentsToDeactivate.length === 0) {
     // Short-circuit to skip reporting telemetry
     return;
   }
 
-  await uninstallExtensionsAndSaveState(toUninstall, {
+  await deactivateModComponentsAndSaveState(modComponentsToDeactivate, {
     editorState,
     optionsState,
   });
 
   reportEvent(Events.DEPLOYMENT_DEACTIVATE_UNASSIGNED, {
     auto: true,
-    deployments: toUninstall.map((x) => x._deployment.id),
+    deployments: modComponentsToDeactivate.map((x) => x._deployment.id),
   });
 }
 
-async function uninstallRecipe(
+async function deactivateMod(
+  modId: RegistryId,
   optionsState: ModComponentState,
   editorState: EditorState | undefined,
-  recipeId: RegistryId,
-): Promise<{
-  options: ModComponentState;
-  editor: EditorState | undefined;
-}> {
+): Promise<{ options: ModComponentState; editor: EditorState | undefined }> {
   let options = optionsState;
   let editor = editorState;
```

Comment:
NIT: `options` and `editor` are ambiguous. 

```suggestion
  let _optionsState = optionsState;
  let _editorState = editorState;
```